src/main/java/us/abstracta/wiresham/Flow.java

```diff
@@ -272,8 +276,15 @@ public int getPortCount() {
 
   public List<Integer> getPorts() {
     return steps.stream()
-        .filter(p -> p instanceof SendPacketStep)
-        .map(PacketStep::getPort)
+        .map(packetStep -> {
+          if (packetStep instanceof SendPacketStep) {
```

Comment:
why sendPackets? What about receive packets? what about include packets? Maybe just a getPorts in any packet and in single packet scenarios returns a singleton list?